tests/e2e/multicluster/multicluster_multiprimary_test.go

```diff
@@ -0,0 +1,479 @@
+					BeforeAll(func(ctx SpecContext) {
+						pushIntermediateCA()
+
+						multiclusterYAML := `
+apiVersion: sailoperator.io/v1alpha1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: %s
+  namespace: %s
+  values:
+    global:
+      meshID: %s
+      multiCluster:
+        clusterName: %s
+      network: %s`
+						multiclusterPrimaryYAML := fmt.Sprintf(multiclusterYAML, version.Name, controlPlaneNamespace, "mesh1", "cluster1", "network1")
+						Log("Istio CR Primary: ", multiclusterPrimaryYAML)
+						Expect(kubectl.CreateFromString(multiclusterPrimaryYAML, kubeconfig)).To(Succeed(), "Istio Resource creation failed on Primary Cluster")
+
+						multiclusterRemoteYAML := fmt.Sprintf(multiclusterYAML, version.Name, controlPlaneNamespace, "mesh1", "cluster2", "network2")
+						Log("Istio CR Remote: ", multiclusterRemoteYAML)
+						Expect(kubectl.CreateFromString(multiclusterRemoteYAML, kubeconfig2)).To(Succeed(), "Istio Resource creation failed on Remote Cluster")
+					})
+
+					It("updates both Istio CR status to Ready", func(ctx SpecContext) {
+						Eventually(common.GetObject).
+							WithArguments(ctx, clPrimary, kube.Key(istioName), &v1alpha1.Istio{}).
+							Should(HaveCondition(v1alpha1.IstioConditionReady, metav1.ConditionTrue), "Istio is not Ready on Primary; unexpected Condition")
+						Success("Istio CR is Ready on Primary Cluster")
+
+						Eventually(common.GetObject).
+							WithArguments(ctx, clRemote, kube.Key(istioName), &v1alpha1.Istio{}).
+							Should(HaveCondition(v1alpha1.IstioConditionReady, metav1.ConditionTrue), "Istio is not Ready on Remote; unexpected Condition")
+						Success("Istio CR is Ready on Primary Cluster")
+					})
+
+					It("deploys istiod", func(ctx SpecContext) {
+						Eventually(common.GetObject).
+							WithArguments(ctx, clPrimary, kube.Key("istiod", controlPlaneNamespace), &appsv1.Deployment{}).
+							Should(HaveCondition(appsv1.DeploymentAvailable, metav1.ConditionTrue), "Istiod is not Available on Primary; unexpected Condition")
+						Expect(common.GetVersionFromIstiod()).To(Equal(version.Version), "Unexpected istiod version")
+						Success("Istiod is deployed in the namespace and Running on Primary Cluster")
+
+						Eventually(common.GetObject).
+							WithArguments(ctx, clRemote, kube.Key("istiod", controlPlaneNamespace), &appsv1.Deployment{}).
+							Should(HaveCondition(appsv1.DeploymentAvailable, metav1.ConditionTrue), "Istiod is not Available on Remote; unexpected Condition")
+						Expect(common.GetVersionFromIstiod()).To(Equal(version.Version), "Unexpected istiod version")
+						Success("Istiod is deployed in the namespace and Running on Remote Cluster")
+					})
+
+				})
+
+				When("Gateway is created in both clusters", func() {
+					BeforeAll(func(ctx SpecContext) {
+						eastGatewayURL := "https://raw.githubusercontent.com/istio-ecosystem/sail-operator/main/docs/multicluster/east-west-gateway-net1.yaml"
```

Comment:
This makes the tests reliant on connectivity to `http://raw.githubusercontent.com`. Maybe we can copy this yaml into a `testdata` folder or construct the yaml inside the tests?

Reply:
I'll check for a better way to do this. Copying this into the config folder can be an option. Construct the yaml also, but I do not want to make the test too difficult to read